File: bonsai_interfaces/src/main/java/de/unibi/citec/clf/btl/data/speechrec/Language.java
package de.unibi.citec.clf.btl.data.speechrec;


import java.util.HashMap;
import java.util.Map;

public enum Language {
    // From Whisper Language codes
    EN((byte) 0),
    ZH((byte) 1),
    DE((byte) 2),
    ES((byte) 3),
    RU((byte) 4),
    KO((byte) 5),
    FR((byte) 6),
    JA((byte) 6),
    PT((byte) 7),
    TR((byte) 8),
    PL((byte) 9),
    CA((byte) 10),
    NL((byte) 11),
    AR((byte) 12),
    SV((byte) 13),
    IT((byte) 14),
    ID((byte) 15),
    HI((byte) 16),
    FI((byte) 17),
    VI((byte) 18),
    HE((byte) 19),
    UK((byte) 20),
    EL((byte) 21),
    MS((byte) 22),
    CS((byte) 23),
    RO((byte) 24),
    DA((byte) 25),
    HU((byte) 26),
    TA((byte) 27),
    NO((byte) 28),
    TH((byte) 29),
    UR((byte) 30),
    HR((byte) 31),
    BG((byte) 32),
    LT((byte) 33),
    LA((byte) 34),
    MI((byte) 35),
    ML((byte) 36),
    CY((byte) 37),
    SK((byte) 38),
    TE((byte) 39),
    FA((byte) 40),
    LV((byte) 41),
    BN((byte) 42),
    SR((byte) 43),
    AZ((byte) 44),
    SL((byte) 45),
    KN((byte) 46),
    ET((byte) 47),
    MK((byte) 48),
    BR((byte) 49),
    EU((byte) 50),
    IS((byte) 51),
    HY((byte) 52),
    NE((byte) 53),
    MN((byte) 54),
    BS((byte) 55),
    KK((byte) 56),
    SQ((byte) 57),
    SW((byte) 58),
    GL((byte) 59),
    MR((byte) 60),
    PA((byte) 61),
    SI((byte) 62),
    KM((byte) 63),
    SN((byte) 64),
    YO((byte) 65),
    SO((byte) 66),
    AF((byte) 67),
    OC((byte) 68),
    KA((byte) 69),
    BE((byte) 70),
    TG((byte) 71),
    SD((byte) 72),
    GU((byte) 73),
    AM((byte) 74),
    YI((byte) 75),
    LO((byte) 76),
    UZ((byte) 77),
    FO((byte) 78),
    HT((byte) 79),
    PS((byte) 80),
    TK((byte) 81),
    NN((byte) 82),
    MT((byte) 83),
    SA((byte) 84),
    LB((byte) 85),
    MY((byte) 86),
    BO((byte) 87),
    TL((byte) 88),
    MG((byte) 89),
    AS((byte) 90),
    TT((byte) 91),
    HAW((byte) 92),
    LN((byte) 93),
    HA((byte) 94),
    BA((byte) 95),
    JW((byte) 96),
    SU((byte) 97),
    YUE((byte) 98);

    private static final Map<Byte, Language> BY_BYTE = new HashMap<>();

    public final Byte value;

    static {
        for (Language e : values()) {
            BY_BYTE.put(e.value, e);
        }
    }

    Language(byte i) {
        this.value = i;
    }

    public static Language valueOf(Byte b) {
        return BY_BYTE.get(b);
    }
}

File: bonsai_interfaces/src/main/java/de/unibi/citec/clf/btl/data/speechrec/Language.kt
package de.unibi.citec.clf.btl.data.speechrec

import de.unibi.citec.clf.btl.Type

data class LanguageType(val value : Language) : Type()

enum class Language(val value: Byte) {
    // From Whisper Language codes
    EN(0.toByte()),
    ZH(1.toByte()),
    DE(2.toByte()),
    ES(3.toByte()),
    RU(4.toByte()),
    KO(5.toByte()),
    FR(6.toByte()),
    JA(7.toByte()),
    PT(8.toByte()),
    TR(9.toByte()),
    PL(10.toByte()),
    CA(11.toByte()),
    NL(12.toByte()),
    AR(13.toByte()),
    SV(14.toByte()),
    IT(15.toByte()),
    ID(16.toByte()),
    HI(17.toByte()),
    FI(18.toByte()),
    VI(19.toByte()),
    HE(20.toByte()),
    UK(21.toByte()),
    EL(22.toByte()),
    MS(23.toByte()),
    CS(24.toByte()),
    RO(25.toByte()),
    DA(26.toByte()),
    HU(27.toByte()),
    TA(28.toByte()),
    NO(29.toByte()),
    TH(30.toByte()),
    UR(31.toByte()),
    HR(32.toByte()),
    BG(33.toByte()),
    LT(34.toByte()),
    LA(35.toByte()),
    MI(36.toByte()),
    ML(37.toByte()),
    CY(38.toByte()),
    SK(39.toByte()),
    TE(40.toByte()),
    FA(41.toByte()),
    LV(42.toByte()),
    BN(43.toByte()),
    SR(44.toByte()),
    AZ(45.toByte()),
    SL(46.toByte()),
    KN(47.toByte()),
    ET(48.toByte()),
    MK(49.toByte()),
    BR(50.toByte()),
    EU(51.toByte()),
    IS(52.toByte()),
    HY(53.toByte()),
    NE(54.toByte()),
    MN(55.toByte()),
    BS(56.toByte()),
    KK(57.toByte()),
    SQ(58.toByte()),
    SW(59.toByte()),
    GL(60.toByte()),
    MR(61.toByte()),
    PA(62.toByte()),
    SI(63.toByte()),
    KM(64.toByte()),
    SN(65.toByte()),
    YO(66.toByte()),
    SO(67.toByte()),
    AF(68.toByte()),
    OC(69.toByte()),
    KA(70.toByte()),
    BE(71.toByte()),
    TG(72.toByte()),
    SD(73.toByte()),
    GU(74.toByte()),
    AM(75.toByte()),
    YI(76.toByte()),
    LO(77.toByte()),
    UZ(78.toByte()),
    FO(79.toByte()),
    HT(80.toByte()),
    PS(81.toByte()),
    TK(82.toByte()),
    NN(83.toByte()),
    MT(84.toByte()),
    SA(85.toByte()),
    LB(86.toByte()),
    MY(87.toByte()),
    BO(88.toByte()),
    TL(89.toByte()),
    MG(90.toByte()),
    AS(91.toByte()),
    TT(92.toByte()),
    HAW(93.toByte()),
    LN(94.toByte()),
    HA(95.toByte()),
    BA(96.toByte()),
    JW(97.toByte()),
    SU(98.toByte()),
    YUE(99.toByte());

    companion object {
        private val BY_BYTE: MutableMap<Byte, Language> = HashMap()

        init {
            for (e in entries) {
                BY_BYTE[e.value] = e
            }
        }

        fun valueOf(b: Byte): Language {
            return BY_BYTE[b] ?: EN
        }
    }
}
